### DISCUSSION TITLE
HTML5, CSS, Javascript. ReactJS? I want to be front end developer, Any guide?

### DISCUSSION THREAD

**Discussion by aanthonytomas** [community/community#126869] [GitHub Education] [answered]
### Hi everyone,

I intermediate at HTML and CSS. Basic only Javascript. Also learned bootstrap. How can I start my journey in React? any tips? and roadmaps?

**Comment by CopperEagle** [comment]
Hey there @aanthonytomas 

If you want to start this journey, then have a look at [this](https://www.freecodecamp.org/news/the-react-roadmap-learn-react/) roadmap. A tutorial is [here](https://react.dev/learn) for starters. Apply your knowledge starting with small projects, scaling them up in complexity over time. A personal webpage is a common start which you can evolve over time and add new features.

When going on a coding adventure, having the [documentation](https://react.dev/reference/react) with you is usually a good idea.

Good luck!

**Reply by TEERTH-KOLHI** [reply]
Very helpful, thanks

**Comment by Balaji2810** [comment, marked as answer]
To become a proficient front-end developer with skills in HTML5, CSS, JavaScript, and ReactJS, follow this structured guide:

### Step-by-Step Guide

#### 1. **Basics of Web Development**
   - **HTML5:** Learn the structure and semantics of HTML5.
     - [HTML5 Tutorial by MDN](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5)
     - [W3Schools HTML5 Tutorial](https://www.w3schools.com/html/default.asp)
   - **CSS:** Understand styling, layout, and design principles.
     - [CSS Tutorial by MDN](https://developer.mozilla.org/en-US/docs/Web/CSS)
     - [W3Schools CSS Tutorial](https://www.w3schools.com/css/default.asp)
   - **JavaScript:** Learn the fundamentals of JavaScript programming.
     - [JavaScript Tutorial by MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript)
     - [W3Schools JavaScript Tutorial](https://www.w3schools.com/js/default.asp)

#### 2. **Advanced Web Development**
   - **CSS Frameworks:** Learn to use frameworks like Bootstrap for responsive design.
     - [Bootstrap Documentation](https://getbootstrap.com/docs/5.0/getting-started/introduction/)
   - **JavaScript ES6:** Get comfortable with modern JavaScript features.
     - [JavaScript ES6 Guide](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/#es6)
   - **Git and GitHub:** Version control and collaboration.
     - [Git Tutorial by Atlassian](https://www.atlassian.com/git/tutorials)
     - [GitHub Guide](https://guides.github.com/activities/hello-world/)

#### 3. **Front-End Frameworks**
   - **ReactJS:** Learn the basics and advanced concepts of ReactJS.
     - [React Official Documentation](https://reactjs.org/docs/getting-started.html)
     - [React Tutorial by FreeCodeCamp](https://www.freecodecamp.org/news/learn-react-js-in-this-free-7-hour-course/)

#### 4. **Building Projects**
   - **Simple Projects:** Start with small projects like a personal portfolio, a to-do list app, or a simple blog.
     - [Project Ideas](https://www.frontendmentor.io/challenges)
   - **Intermediate Projects:** Build more complex applications like an e-commerce site, a weather app, or a chat application.
     - [Project Ideas and Tutorials](https://www.fullstackopen.com/en/)

#### 5. **Learning Resources**
   - **Online Courses:**
     - [FreeCodeCamp](https://www.freecodecamp.org/)
     - [Codecademy](https://www.codecademy.com/catalog/subject/web-development)
     - [Udemy (JavaScript and React courses)](https://www.udemy.com/)
   - **Books:**
     - [Eloquent JavaScript](https://eloquentjavascript.net/)
     - [You Don’t Know JS (book series)](https://github.com/getify/You-Dont-Know-JS)
   - **Blogs and Articles:**
     - [CSS-Tricks](https://css-tricks.com/)
     - [Smashing Magazine](https://www.smashingmagazine.com/)

#### 6. **Practice and Improve**
   - **Coding Challenges:** Participate in coding challenges to sharpen your skills.
     - [HackerRank](https://www.hackerrank.com/domains/tutorials/10-days-of-javascript)
     - [LeetCode](https://leetcode.com/explore/featured/card/10-days-of-javascript/)
   - **Join Communities:** Engage with the developer community to get support and feedback.
     - [Stack Overflow](https://stackoverflow.com/)
     - [Dev.to](https://dev.to/)

#### 7. **Portfolio and Job Search**
   - **Create a Portfolio:** Showcase your projects and skills.
     - [Creating a Developer Portfolio](https://careerfoundry.com/en/blog/web-development/how-to-build-a-web-developer-portfolio/)
   - **Apply for Jobs:** Use job boards and networking to find opportunities.
     - [LinkedIn](https://www.linkedin.com/)
     - [Indeed](https://www.indeed.com/)

### Helpful Links
- [HTML5 MDN](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5)
- [CSS MDN](https://developer.mozilla.org/en-US/docs/Web/CSS)
- [JavaScript MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript)
- [ReactJS Documentation](https://reactjs.org/docs/getting-started.html)
- [FreeCodeCamp](https://www.freecodecamp.org/)
- [Git Tutorial](https://www.atlassian.com/git/tutorials)

By following this guide, you will develop a solid foundation in front-end development and build the skills needed to become a proficient developer. Good luck!
